plugins/callback/opentelemetry.py

```diff
@@ -0,0 +1,445 @@
+# (C) 2021, Victor Martinez <VictorMartinezRubio@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    author: Victor Martinez (@v1v)  <VictorMartinezRubio@gmail.com>
+    name: opentelemetry
+    type: notification
+    short_description: Create distributed traces with OpenTelemetry
+    version_added: 2.5.0
+    description:
+      - This callback create distributed traces for each Ansible task with OpenTelemetry.
+      - You can configure the OTEL exporter with environment variables. See U(https://opentelemetry-python.readthedocs.io/en/latest/exporter/otlp/otlp.html).
+    options:
+      hide_task_arguments:
+        default: false
+        description:
+          - Hide the arguments for a task.
+        env:
+          - name: OPENTELEMETRY_HIDE_TASK_ARGUMENTS
+      console_output:
+        default: false
+        description:
+          - Print distributed traces in the terminal.
+        env:
+          - name: OPENTELEMETRY_CONSOLE_OUTPUT
+      otel_service_name:
+        default: ansible
+        description:
+          - The service name resource attribute.
+        env:
+          - name: OTEL_SERVICE_NAME
+    requirements:
+      - opentelemetry-api (python lib)
+      - opentelemetry-exporter-otlp (python lib)
+      - opentelemetry-sdk (python lib)
+'''
+
+
+EXAMPLES = '''
+examples: |
+  Whitelist the plugin in ansible.cfg:
+    [defaults]
+    callback_whitelist = community.general.opentelemetry
+
+  Set the environment variable:
+    export OTEL_EXPORTER_OTLP_ENDPOINT=<your endpoint (OTLP/HTTP)>
+    export OTEL_EXPORTER_OTLP_HEADERS="authorization=Bearer your_otel_token"
+    export OTEL_SERVICE_NAME=your_service_name
+'''
+
+import getpass
+import os
+import socket
+import sys
+import time
+import uuid
+
+from os.path import basename
+
+from ansible import constants as C
+from ansible.plugins.callback import CallbackBase
+
+try:
+    from opentelemetry import trace
+    from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter
+    from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+    from opentelemetry.trace.status import Status, StatusCode
+    from opentelemetry.sdk.trace import TracerProvider
+    from opentelemetry.sdk.trace.export import (
+        ConsoleSpanExporter,
+        SimpleSpanProcessor,
+        BatchSpanProcessor
+    )
+    from opentelemetry.util._time import _time_ns
+    HAS_OTEL = True
+except ImportError:
+    HAS_OTEL = False
```

Comment:
You might also want to catch the exception and provide it to the user; see https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/dev_guide/testing/sanity/import.rst for how to do that.